app/models/host/managed.rb

```diff
@@ -926,6 +927,17 @@ def apply_compute_profile(modification)
 
   private
 
+  def owner_taxonomies_match
+    if self.owner.is_a?(User)
+      if Organization.organizations_enabled && self.organization_id && !self.owner.my_organizations.where(:id => self.organization_id).exists?
```

Comment:
`self.owner.organization_and_child_ids.include?(self.organization_id)` so you can avoid adding new methods in user.rb, location.rb and organization.rb

Why not implementing these through `Taxonomy.enabled_taxonomies` by the way? It's basically duplicated code as is now